packages/components/src/internal/components/labels/BarTenderSettingsForm.tsx

```diff
@@ -101,173 +92,149 @@ const SettingsInput: FC<SettingsInputProps> = memo(({ children, description, lab
     );
 });
 
-// exported for jest testing
-export class BarTenderSettingsFormImpl extends PureComponent<Props, State> {
-    static defaultProps = {
-        api: getDefaultAPIWrapper(),
-    };
-
-    private btTestConnectionTemplate = (label: string): string => {
-        // Should be able to run connection test w/o a default label set.
-        const formatNode = `<Format>${label}</Format>`;
+const btTestConnectionTemplate = (label: string): string => {
+    // Should be able to run connection test w/o a default label set.
+    const formatNode = `<Format>${label}</Format>`;
 
-        return `<XMLScript Version="2.0">
+    return `<XMLScript Version="2.0">
             <Command Name="Job1">
                 <FormatSetup>
                     ${formatNode}
                 </FormatSetup>
             </Command>
         </XMLScript>`;
-    };
-
-    constructor(props: Props) {
-        super(props);
-
-        this.state = {
-            btServiceURL: undefined,
-            dirty: false,
-            submitting: false,
-            testing: false,
-            saveErrorMsg: undefined,
-            connectionValidated: undefined, // Don't know if it has been verified
-            failureMessage: undefined,
-        };
-    }
-
-    componentDidMount(): void {
-        this.load();
-    }
-
-    load = (): void => {
-        this.props.api.labelprinting.fetchBarTenderConfiguration().then(btConfiguration => {
-            this.setState(() => ({
-                btServiceURL: btConfiguration.serviceURL,
-                dirty: false,
-                submitting: false,
-            }));
+};
+const apiWrapper = getDefaultAPIWrapper();
+// exported for jest testing
+export const BarTenderSettingsFormImpl: FC<Props> = memo(props => {
+    const { api = apiWrapper, title = BARTENDER_CONFIGURATION_TITLE, onChange, onSuccess } = props;
+    const [btServiceURL, setBtServiceURL] = useState<string>();
+    const [defaultLabel, setDefaultLabel] = useState<number>();
+    const [dirty, setDirty] = useState<boolean>(false);
+    const [submitting, setSubmitting] = useState<boolean>(false);
+    const [testing, setTesting] = useState<boolean>();
+    const [connectionValidated, setConnectionValidated] = useState<boolean>();
+    const [failureMessage, setFailureMessage] = useState<string>();
+    const [loading, setLoading] = useState<boolean>(true);
+
+    useEffect(() => {
+        api.labelprinting.fetchBarTenderConfiguration().then(btConfiguration => {
```

Comment:
does this API call need a `.catch()` to handle the error case?

Reply:
added, but I don't think so